pkg/eval/ievaluator.go

```diff
@@ -26,3 +40,11 @@ type IEvaluator interface {
 		flagKey string,
 		context *structpb.Struct) (value map[string]any, variant string, reasons string, err error)
 }
+
+func (s *StateChangeNotification) ToMap() map[string]interface{} {
+	return map[string]interface{}{
```

Comment:
Do we need some guards here on s.type?

Reply:
s.Type is a typed string of `StateChangeNotificationType`, it needs to be cast into a plain string type to allow for the structpb library to handle it. This type conversion will never fail.